code/modules/games/cards/lotto.dm

```diff
@@ -0,0 +1,131 @@
+		if(!istype(src,/obj/item/toy/lotto_ticket/supermatter_surprise))
+			return
+
+//Flash code taken from Blinder
+/obj/item/toy/lotto_ticket/proc/flash(var/turf/T , var/mob/living/M)
+	playsound(src, 'sound/effects/EMPulse.ogg', 100, 1)
+
+	if(M.blinded)
+		return
+
+	M.flash_eyes(visual = 1, affect_silicon = 1)
+
+	if(issilicon(M))
+		M.Knockdown(rand(5, 10))
+		M.visible_message("<span class='warning'>[M]'s sensors are overloaded by the flash of light!</span>","<span class='warning'>Your sensors are overloaded by the flash of light!</span>")
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		var/datum/organ/internal/eyes/E = H.internal_organs_by_name["eyes"]
+		if (E && E.damage >= E.min_bruised_damage)
+			to_chat(M, "<span class='warning'>Your eyes start to burn badly!</span>")
+	M.update_icons()
+
+
+/obj/item/toy/lotto_ticket/attackby(obj/item/weapon/S as obj, mob/user as mob)
+	if(!src.revealed == 1)
+		if(S.is_sharp() || istype(S, /obj/item/weapon/coin))
+			if(do_after(user, src, 1 SECONDS))
+				src.revealed = 1
+				src.update_icon()
+				to_chat(user, "<span class='notice'>You scratch off the film covering the prizes.</span>")
+				if(istype(src,/obj/item/toy/lotto_ticket/supermatter_surprise))
+					to_chat(user, "<span class='notice'>Removing the film emits a brilliant flash of light!</span>")
+					//Radiation emission code taken from Jukebox
+					emitted_harvestable_radiation(get_turf(src), 20, range = 5)	//Standing by a juke applies a dose of 17 rads to humans so we'll round based on that. 1/5th the power of a freshly born stage 1 singularity.
+					for(var/mob/living/carbon/M in view(src,3))
+						var/rads = 50 * sqrt( 1 / (get_dist(M, src) + 1) ) //It's like a transmitter, but 1/3 as powerful.
+						M.apply_radiation(round(rads/2),RAD_EXTERNAL) //Distance/rads: 1 = 18, 2 = 14, 3 = 12
+					var/flash_turf = get_turf(src)
+					for(var/mob/living/M in get_hearers_in_view(3, flash_turf))
+						flash(get_turf(M), M)
+				winnings = scratch(ticket_price)
+				if(winnings > 0)
+					src.iswinner = 1
+				return
+		else
+			to_chat(user, "<span class='notice'>You need to use something sharp to scratch the ticket.</span>")
+			return
+	else
+		to_chat(user, "<span class='notice'>The film covering the prizes has already been scratched off.</span>")
+		return
+
+/obj/item/toy/lotto_ticket/examine(mob/user)
+	if(user.range_check(src))
+		if(revealed == 1)
+			..()
+			if(iswinner == 1)
+				to_chat(user, "<span class='notice'>This one is a winner! You've won [winnings] credits.</span>")
+			else
+				to_chat(user, "<span class='notice'>No wins on this one.</span>")
+		else
+			..()
+			to_chat(user, "<span class='notice'>It hasn't been scratched off yet.</span>")
+	else
+		..() //Only show a regular description if it is too far away to read.
+		to_chat(user, "<span class='notice'>It is too far away to read.</span>")
+
+/obj/item/toy/lotto_ticket/update_icon()
+	if(istype(src,/obj/item/toy/lotto_ticket/gold_rush))
```

Comment:
avoid istype checks against src, instead you can do this all in one line...

`icon_state = initial(icon_state) + (revealed ? "_scratched" : "")`

This also means it is safe to call update_icon even if the ticket isn't revealed, it will always set the correct icon with that ternary statement